src/libYARP_OS/include/yarp/os/Port.h

```diff
@@ -130,6 +130,13 @@ class YARP_OS_API Port : public UnbufferedContactable
     // Documented in Contactable
     void setAdminReader(PortReader& reader) override;
 
+    /**
+     * Subscribe to port reader new state with given callback.
+     *
+     * @param callback can be provided for example in lambda format [](ConnectionReader& conn){}
```

Comment:
I think you should add some documentation explaining that setReader and setReaderCallback are mutually exclusive, and by calling one of the two methods the previous callback is removed.